src/NumericalAlgorithms/Interpolation/RegularGridInterpolant.cpp

```diff
@@ -0,0 +1,80 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "RegularGridInterpolant.hpp"
+
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Mesh.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace intrp {
+
+template <size_t Dim>
+RegularGrid<Dim>::RegularGrid(
+    const Mesh<Dim>& source_mesh, const Mesh<Dim>& target_mesh,
+    const std::array<DataVector, Dim>&
+        override_target_mesh_with_1d_logical_coords) noexcept
+    : number_of_target_points_(1), source_extents_(source_mesh.extents()) {
+  for (size_t d = 0; d < Dim; ++d) {
+    const auto source_mesh_1d = source_mesh.slice_through(d);
+    const auto target_mesh_1d = target_mesh.slice_through(d);
+    if (gsl::at(override_target_mesh_with_1d_logical_coords, d).size() == 0) {
+      // Normal mode: use target_mesh
+      // Only make a matrix if source and target meshes differ; when a default-
+      // constructed matrix is given instead, apply_matrices does no work.
+      if (source_mesh_1d != target_mesh_1d) {
+        gsl::at(interpolation_matrices_, d) = Spectral::interpolation_matrix(
+            source_mesh_1d, get<0>(logical_coordinates(target_mesh_1d)));
+      }
+    } else {
+      // Override mode: override target_mesh with given points
+      gsl::at(interpolation_matrices_, d) = Spectral::interpolation_matrix(
+          source_mesh.slice_through(d),
+          gsl::at(override_target_mesh_with_1d_logical_coords, d));
+    }
+    number_of_target_points_ *= gsl::at(interpolation_matrices_, d).rows();
+  }
+}
+
+template <size_t Dim>
+RegularGrid<Dim>::RegularGrid() = default;
+
+template <size_t Dim>
+void RegularGrid<Dim>::pup(PUP::er& p) noexcept {
+  p | number_of_target_points_;
+  p | source_extents_;
+  p | interpolation_matrices_;
+}
+
+template <size_t LocalDim>
+bool operator==(const RegularGrid<LocalDim>& lhs,
+                const RegularGrid<LocalDim>& rhs) noexcept {
+  return lhs.number_of_target_points_ and rhs.number_of_target_points_ and
```

Comment:
Close, I think you want the first `and` to be an `==` :)

Reply:
self-rekt :(